ARCs/arc-0060.md

```diff
@@ -0,0 +1,449 @@
+}
+```
+
+`SignDataError` uses the same error codes of `SignTxnsError` as well as the following codes:
+
+| Status Code | Name | Description |
+| --- | --- | --- |
+| 4600 | Invalid scope | The scope of the signing action is not recognized by the wallet |
+| 4601 | Invalid schema | The schema does not comply with ARC-60 requirements |
+| 4602 | Invalid encoding | Data cannot be decoded with the given encoding |
+
+### Semantic Requirements
+
+The call `signData(signingData, metadata)` **MUST** either return the signed data `ret` or reject the call throwing an error `err`.
+
+> Following [ARC-1](./arc-0001.md) terminology, in this ARC the term **Rejecting** means throwing an error with `err.code=4300`.
+
+Upon calling `signData(signingData,, metadata)`:
+
+- the `signingData` and `metadata` **MUST NOT** be `null`, otherwise the wallet **MUST** reject the call.
+- `signingData.data` **MUST** be valid respect the JSON schema in `metadata.schema`. If the validation fails, the call **MUST** be rejected with a `4300` error.
+- if the encoding `metadata.encoding` is present, it **MUST** be used to decode `signingData.data.bytes`.
+- the wallet **MUST** prepare the signing message as `msg=(<signingData.data.bytes>)`.
+  - if the value of `ARC60Domain` is not empty than it **MUST** be used as `prefix`: `msg=(<prefix> || <signingData.data.bytes>)`.
+- the wallet **MUST** ask users for signing confirmation. It **MUST** display a message (defined in Display message) providing the `ScopeType`, `signer`, and the `msg` being signed:
+  - if the user approves, `msg` **MUST** be signed with the `signer`'s key and `ret` **MUST** be set to the corresponding `Signature`.
+  - if the user rejects, the call **MUST** fail with error code `4001`.
+
+Note that if `msg` cannot be signed for any reason, the wallet **MUST** throw an error, such that
+
+- `err.message` **SHOULD** indicate the reason for the error (e.g. specify that `signingData.data` is not a valid JSON object according to `metadata.schema`)
+- `err.failingSignData` **SHOULD** return the `StdSigData` object that caused the error, otherwise `null`.
+
+#### Semantic of `StdSigData`
+
+- it **MUST** be a valid `StdSigData` object, otherwise the wallet **MUST** reject.
+- `data`:
+  - it **MUST** be a valid `StdDataStr` object (UTF-8 encoded), otherwise the wallet **MUST** reject.
+  - it **MUST** adhere to `metadata.schema`, otherwise the wallet **MUST** reject.
+  - the `data.ARC60Domain` field must be validated against forbidden values:
+    - the wallet **MUST** reject if `ARC60Domain="TX"`.
+    - the wallet **MUST** reject if `ARC60Domain="TG"`.
+  - `data.bytes` must be a valid byte string:
+    - it **MUST** be encoded as `metadata.encoding` if present, otherwise `base64`.
+    - the wallet **MUST** throw a `4602` error if the decoding fails with the given encoding value.
+    - the wallet **MUST** reject if `data.bytes` field is prepended with a forbidden domain separator `TX` or `TG`.
```

Comment:
Just marking this for completeness -- same as above

Reply:
https://github.com/algorandfoundation/ARCs/pull/284/commits/19720ff8a4b186fedd7f9884a3d97400d6f94780